crates/katana/core/src/messaging/service.rs

```diff
@@ -0,0 +1,325 @@
+//! Messaging service is a Future which is polling two streams,
+//! one for gathering the messages from the settlement chain,
+//! and an other one to settle messages on the settlement chain.
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+use std::time::Duration;
+
+use futures::{Future, FutureExt, Stream, StreamExt};
+use starknet::core::types::{FieldElement, MsgToL1};
+use tokio::sync::RwLock as AsyncRwLock;
+use tokio::time::{interval_at, Instant, Interval};
+use tracing::{error, info, trace};
+
+use crate::backend::storage::transaction::{L1HandlerTransaction, Transaction};
+use crate::backend::Backend;
+use crate::messaging::starknet_messenger::HASH_EXEC;
+use crate::messaging::{AnyMessenger, MessagingConfig, Messenger, MSGING_TARGET};
+use crate::pool::TransactionPool;
+
+pub struct MessageService {
+    settler: MessageSettler,
+    gatherer: MessageGatherer,
+}
+
+impl MessageService {
+    /// Initializes a new instance from a configuration file's path.
+    /// Will panic on failure to avoid continuing with invalid configuration.
+    pub async fn new(
+        config_path: std::path::PathBuf,
```

Comment:
```suggestion
        config: MessagingConfig,
```

prefer for this to accept `MessagingConfig` direclty instead of a path